File: 2020-week1/c++/2020-week1_5.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<long long int> fibo(n + 1, 0);
    if (n <= 1)
    {
        fibo[n] = n;
    }
    else
    {
        fibo[1] = 1;
        for (int i = 2; i <= n; i++)
        {
            fibo[i] = fibo[i - 1] + fibo[i - 2];
        }
    }

    cout << fibo[n] << endl;
    return 0;
}

File: 2020-week1/java/2020-week1_5.java
import java.util.Scanner;

public class Fibonacci_2748 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		
		long[] fibo = new long[n+1];
		fibo[0]=0;
		for(int i=1; i<n+1; i++) {
			if(i==1) {
				fibo[i] = 1;
				continue;
			}
			fibo[i] = fibo[i-1] + fibo[i-2];
		}
		
		System.out.println(fibo[n]);
	}
}